src/linux/epoll.rs

```diff
@@ -301,6 +301,7 @@ impl Epoll {
         timeout: i32,
         events: &mut [EpollEvent],
```

Comment:
Why do we keep `events` slice decoupled from `max_events`?

Is there a usability argument  against dropping `max_events` as an explicit parameter and just using `events.len()` in the code?

Reply:
The use case for which we keep `events.len()` decoupled from `max_events` is that `events` can be allocated once, and reused multiple times in different scenarios. Say we are initializing events with a capacity of 256, but in some cases we just want to return 10 events, and in others 256. With this is approach, `events` can be reused.

The other reason why we went on this path, is because we tried to keep `vmm-sys-util` as a minimal, safe wrapper of existing utilities, so we just keep the same parameters as the inner functions we are calling.

This is not really documented, so it can be a bit confusing. Sorry about that.

Reply:
There are indeed some unofficial links, e.g. https://copyconstruct.medium.com/the-method-to-epolls-madness-d9d2d6378642, that state the fact that `max_events` is the `events` len and in general this is how you will use `epoll_wait`, but I wouldn't add this restriction to our API. The error throwing fix seems enough to me.

Reply:
The current API looks very un-rusty and very c-like (from what you said, that's probably by design).

I think the better UX is to use `slice::len()` since in Rust a slice is a view in a buffer consisting of idx and len. The `max_events` param is totally extraneous in a Rust function.

It all comes down to using the full power of Rust slices:
> Say we are initializing events with a capacity of 256, but in some cases we just want to return 10 events, and in others 256. With this is approach, events can be reused.

You could do that idiomatically:
```diff
- let ev_count = epoll.wait(10, -1, &mut ready_events[..]).unwrap();
+ let ev_count = epoll.wait(-1, &mut ready_events[..10]).unwrap();
```

Reply:
Adding the extraneous param just increases chances of misuse like seen here.

Adding the explicit sanity check for `max_events` and throwing an error also obviously works but moves _mistakes_ discovery from compile-time to runtime.

These are just my 2 cents, don't read these comments as blockers to the PR.

Reply:
Sounds good :+1:

Reply:
I really like this suggestion to remove the argument alltogether. @lauralt if you want to allocate a large `events` block but only fill part of it, then you can just slice it as described by @acatangiu. Can I make this change to remove the `max_events` parameter?

Reply:
Yeah, I got it from the beginning, sorry if it wasn't that clear :D,  I was actually thinking if there could be use cases where you pass a sub-slice as described by @acatangiu, but `max_events` still could be smaller than the len of that subslice. Anyway, after more thinking, I agree the safest and the most rustacean solution is to remove that field. Yep, you can go ahead :+1: